server/routes/products/searchProducts.js

```diff
@@ -46,45 +49,66 @@ router.get("/products", (req, res) => {
         "variant_images.image_url"
       )
       .where((builder) => {
-        if (req.query.category_id) {
+        if (category_id) {
           builder
             .where("c.id", category_id)
             .andWhere(
               "products.name",
               "ILIKE",
-              `%${req.query.keyword.toLowerCase()}%`
+              `%${keyword?.toLowerCase()}%`
             );
         } else {
-          builder
-            .where(
-              "products.name",
-              "ILIKE",
-              `%${req.query.keyword.toLowerCase()}%`
-            )
-            .orWhere(
-              "c.name",
-              "ILIKE",
-              `%${req.query.keyword.toLowerCase()}%`
-            );
+          if (parentCategory.length > 0) {
+            builder
+              .where(
+                "products.name",
+                "ILIKE",
+                `%${keyword?.toLowerCase()}%`
+              )
+              .orWhere("c.parent_id", parentCategory[0].id)
+              .orWhere(
+                "c.name",
+                "ILIKE",
+                `%${keyword?.toLowerCase()}%`
+              );
+          } else {
+            builder
+              .where(
+                "products.name",
```

Comment:
this iwhere query seems to be too big, it has if else condition that duplicates query
Can we reduce that?